design/controlling-agent-updates.md

```diff
@@ -0,0 +1,66 @@
+# Controlling agent updates
+
+The agent is considered an extension of the service.
+As such, we have an auto-update mechanism to ensure that new features can rely on the agent support they require.
+Conceptually, it's no different from binaries on the app tier or JavaScript served to the web frontend.
+
+For instance, tasks express a minimum agent version required.
+If a selected agent doesn't meet that requirement, the agent is first asked to update itself.
+Other ways an agent can be updated include using a feature which demands an evergreen agent (YAML pipelines, for instance) or clicking the "upgrade all agents" button in a pool.
+
+For some highly-regulated customers, this kind of evergreen infrastructure is at odds with their compliance needs.
+Such customers today are doing things like ACLing the agent's installation directory so it can't update itself or recompiling the agent without update support.
+This leaves us in a situation where no one is happy:
+we can't service the product correctly, and the customer has to maintain custom configuration / infrastructure to meet their compliance needs.
+
+## Solution
+
+_See below for other solutions considered._
+
+We will add a mode to the project-level pool which puts it into "no update" mode.
+In this mode, jobs will use the minimum agent version as a demand.
+If no agent can satisfy the demand, the job will fail instead of forcing an agent update.
+
+The error message must clearly indicate what feature or task demanded agent update.
+It must also indicate that the customer opted into this behavior.
+This way, the customer can decide whether to update their agents or edit the pipeline to remove the offending feature.
+For example:
+> This pool doesn't allow agent updates.
+> An administrator can alter this setting on the project pool page.
+> Task `MyTask` version 1.2.3 requires agent version 2.175.1, which could not be satisfied.
+
+Manual agent updates (i.e. clicking the button in the web) should still send update requests as normal.
+
+## Potential problems & solutions
```

Comment:
I think the Pool/Queue UI should make it clear what version each agent is on and ideally flag any agents that have not been updated for any reason.

Reply:
Agree that they should show the agent version (I thought we already did, but we don't). We could also denote any agents that aren't on the latest-according-to-the-AT version, but that's not necessarily a problem so I wouldn't want the note to be too alarming.